cmd/mdatagen/internal/templates/metrics.go.tmpl

```diff
@@ -120,24 +123,45 @@ func (m *metric{{ $name.Render }}) init() {
 
 func (m *metric{{ $name.Render }}) recordDataPoint(start pcommon.Timestamp, ts pcommon.Timestamp, val {{ $metric.Data.MetricValueType.BasicType }}
 {{- range $metric.Attributes -}}{{- if not (attributeInfo .).Optional -}}, {{ .RenderUnexported }}AttributeValue {{ (attributeInfo .).Type.Primitive }}{{ end }}{{ end }}{{- if $metric.HasOptionalAttribute $.Attributes -}}, options ...MetricAttributeOption{{- end -}}) {
-	if !m.config.Enabled {
-		return
-	}
-	dp := m.data.{{ $metric.Data.Type }}().DataPoints().AppendEmpty()
+  if !m.config.Enabled {
+    return
+  }
+  nm := pmetric.NewMetric()
```

Comment:
Why do we need create the metric again? It's supposed to be created by `init()`. We should keep using `m.data.{{ $metric.Data.Type }}()`